hpedockerplugin/hpe_storage_api.py

```diff
@@ -456,6 +580,227 @@ def volumedriver_create(self, name, opts=None):
+                self.hpeplugin_driver.create_cloned_volume(clone_vol, src_vol)
+
+                response = json.dumps({u"Err": ''})
+                # For now just track volume to uuid mapping internally
+                # TODO: Save volume name and uuid mapping in etcd as well
+                # This will make get_vol_byname more efficient
+                self._etcd.save_vol(clone_vol)
+                return response
+            except exception.HPEPluginEtcdException as ex:
+                # TODO: 3PAR clean up issue over here - clone got created
+                # in the backend but since it could not be saved in etcd db
+                # we are throwing an error saying operation failed.
+                # TODO: This needs to be fixed
+                response = json.dumps({u"Err": ex.message})
+                return response
+
+            except Exception as ex:
+                msg = (_('clone volume failed, error is: %s'),
+                       six.text_type(ex))
+                LOG.error(msg)
+                response = json.dumps({u"Err": six.text_type(ex)})
+                return response
+        except exception.HPEPluginEtcdException as ex:
+            # Imran: Returning good response even when exception is caught???
+            response = json.dumps({u"Err": ''})
+            return response
+        except Exception as ex:
+            msg = (_('unknown exception caught while cloning volume '
+                     '%(name)s - reason: %(reason)s',
+                     {'name': clone_name, 'reason': ex.message}))
+            LOG.debug(msg)
+            response = json.dumps({u"Err": ''})
+            return response
+
+        finally:
+            # Release lock and return
+            # NOTE: if for some reason unlock fails, we'll lose this
+            # create exception.
+            # Expand lock code inline as function based lock causes
+            # unexpected behavior
+            if src_lock_acquired:
+                try:
+                    self._etcd.try_unlock_volname(src_vol_name)
+                except Exception as ex:
+                    LOG.debug('volume: %(name)s Unlock Volume Failed',
+                              {'name': src_vol_name})
+                    # response = json.dumps({u"Err": six.text_type(ex)})
+                    # return response
+            if clone_lock_acquired:
+                try:
+                    self._etcd.try_unlock_volname(clone_name)
+                except Exception as ex:
+                    LOG.debug('volume: %(name)s Unlock Volume Failed',
+                              {'name': clone_name})
+                    # response = json.dumps({u"Err": six.text_type(ex)})
+                    # return response
+
+    def volumedriver_create_snapshot(self, name, opts=None):
+        # Repeating the validation here in anticipation that when
+        # actual REST call for snapshot creation is added, this
+        # function will have minimal impact

```

Comment:
Add a LOG.info() line printing the passed parameters and to know that the create snapshot call is being invoked.